tests/test_client_session.py

```diff
@@ -706,3 +706,73 @@ async def test_requote_redirect_url_default_disable() -> None:
     session = ClientSession(requote_redirect_url=False)
     assert not session.requote_redirect_url
     await session.close()
+
+
+@pytest.mark.parametrize(
+    ("base_url", "url", "expected_url"),
+    [
+        pytest.param(
+            None,
+            "http://example.com/test",
+            URL("http://example.com/test"),
+            id="base_url(None) + str(http://example.com/test)",
```

Comment:
I think I'd go for object reprs like this
```suggestion
            id="base_url=None url='http://example.com/test'",
```